src/Service.Tests/Configuration/TelemetryTests.cs

```diff
@@ -0,0 +1,234 @@
+
+    private const string CONFIG_WITH_TELEMETRY = "dab-telemetry-test-config.json";
+    private const string CONFIG_WITHOUT_TELEMETRY = "dab-no-telemetry-test-config.json";
+    private static RuntimeConfig _configuration;
+
+    /// <summary>
+    /// Creates runtime config file with specified telemetry options.
+    /// </summary>
+    /// <param name="configFileName">Name of the config file to be created.</param>
+    /// <param name="isTelemetryEnabled">Whether telemetry is enabled or not.</param>
+    /// <param name="telemetryConnectionString">Telemetry connection string.</param> 
+    public static void SetUpTelemetryInConfig(string configFileName, bool isTelemetryEnabled, string telemetryConnectionString)
+    {
+        DataSource dataSource = new(DatabaseType.MSSQL,
+            GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL), Options: null);
+
+        _configuration = InitMinimalRuntimeConfig(dataSource, graphqlOptions: new(), restOptions: new());
+
+        TelemetryOptions _testTelemetryOptions = new(new ApplicationInsightsOptions(isTelemetryEnabled, telemetryConnectionString));
+        _configuration = _configuration with { Runtime = _configuration.Runtime with { Telemetry = _testTelemetryOptions } };
+
+        File.WriteAllText(configFileName, _configuration.ToJson());
+    }
+
+    /// <summary>
+    /// Cleans up the test environment by deleting the runtime config with telemetry options.
+    /// </summary>
+    [TestCleanup]
+    public void CleanUpTelemetryConfig()
+    {
+        File.Delete(CONFIG_WITH_TELEMETRY);
+        File.Delete(CONFIG_WITHOUT_TELEMETRY);
+        Startup.AppInsightsOptions = new();
+        Startup.CustomTelemetryChannel = null;
+    }
+
+    /// <summary>
+    /// Test for non-hosted scenario.
+    /// Tests that different telemetry items such as Traces or logs, Exceptions and Requests
+    /// are correctly sent to application Insights when enabled.
+    /// Also asserting on their respective properties.
+    /// </summary>
+    /// <note>
+    /// Commenting Assert on Request Telemetry as it is flaky, sometimes passing sometimes failing.
+    /// while on manual testing it is working fine and we see all the request telemetryItems in Application Insights.
+    /// Issue to track the fix for this test: https://github.com/Azure/data-api-builder/issues/1734
+    [TestMethod]
+    public async Task TestTelemetryItemsAreSentCorrectly_NonHostedScenario()
+    {
+        SetUpTelemetryInConfig(CONFIG_WITH_TELEMETRY, isTelemetryEnabled: true, TEST_APP_INSIGHTS_CONN_STRING);
+
+        string[] args = new[]
+        {
+            $"--ConfigFileName={CONFIG_WITH_TELEMETRY}"
+        };
+
+        ITelemetryChannel telemetryChannel = new CustomTelemetryChannel()
+        {
+            EndpointAddress = "https://localhost/"
+        };
+        Startup.CustomTelemetryChannel = telemetryChannel;
+        using (TestServer server = new(Program.CreateWebHostBuilder(args)))
```

Comment:
Is the telemetry client being disposed off correctly - so that the Flush gets called?

Reply:
The TelemetryClient class does not implement the IDisposable interface, which means that it does not need to be explicitly disposed of.

But if you mean the server, i think the using statement will make sure it's disposed off.